models/CustomContentContainer.php

```diff
@@ -190,7 +193,7 @@ public function defaultRules()
             ['target', 'validateTarget'],
             [['type'], 'validateContentType'],
             [['visibility'], 'integer', 'min' => static::VISIBILITY_PRIVATE, 'max' => static::VISIBILITY_ADMIN_ONLY],
-            [['title', 'target'], 'string', 'max' => 255],
+            [['title', 'target', 'iframe_attrs'], 'string', 'max' => 255],
```

Comment:
It is not sufficient to show input only for SystemAdmins and hide it otherwise. An attacker could still transmit the value and it would still be stored.

Solution for this would be to add in the `defaultRules` the attribute `frame_attrs` only for `isAdmin()`. Then it will be considered Unsafe by Yii2 in other cases and not saved.

Reply:
Indeed! What do you think doing it this way? https://github.com/humhub/custom-pages/pull/246/commits/484781def0d102538e1c4e26eb399fde93898baf

I also disabled the possibility for a non admin to change the URL if iframe attributes are present.

Reply:
@funkycram If no rule is defined for an attribute, it is considered unsafe by Yii and will not be loaded. I.e. if we create a rule only for admins, only an admin could change the field. This would have the advantage that later also non-admins can change the IFrame, but not the field.

Example:

```php
    public function defaultRules()
    {
        $result = [
            [['type', 'title', 'target'], 'required'],
            [['type'], 'integer'],
            ['target', 'validateTarget'],
            [['type'], 'validateContentType'],
            [['visibility'], 'integer', 'min' => static::VISIBILITY_PRIVATE, 'max' => static::VISIBILITY_ADMIN_ONLY],
            [['title', 'target'], 'string', 'max' => 255],
        ];

        if (Yii::$app->user->isAdmin() && $this->type === IframeType::ID) {
            // Allow System Admins to modify the Iframe Attrs
            $result[] = [['iframe_attrs'], 'string', 'max' => 255];
        } 

        $result = array_merge($result, $this->getRulesByTarget());
        return array_merge($result, $this->getRulesByContentType());
    }

```

What do you think of the solution?  Or am I missing something here?

Reply:
OK, I understand.

But this solution doesn't prevents non system admin to change the URL of the embedded page if an iframe attribute already exists, what was possible with [the `validateIsAdmin` function](https://github.com/humhub/custom-pages/commit/484781def0d102538e1c4e26eb399fde93898baf#diff-3efc7636e2cf4923ac42d418dae03019e1d4d2ae6be379e37023c2e8f93ff3b8R208) that check if `$this->iframe_attrs` has a value for both attributes (['page_content' and 'iframe_attrs'](https://github.com/humhub/custom-pages/commit/484781def0d102538e1c4e26eb399fde93898baf#diff-3efc7636e2cf4923ac42d418dae03019e1d4d2ae6be379e37023c2e8f93ff3b8R197)).

But if you think the solution you suggest is better, I'll change it for it, it has the advantage to be more simple.

Reply:
I would prefer my approach, because it's a bit smaller and only the potential risky field 'iframe_attrs'  is ignored for non-admins. But's your choice :-)

Reply:
Yes, I agree, simpler is better, and the risk is really small!
I changed it to your solution: https://github.com/humhub/custom-pages/pull/246/commits/da8ec80578f8d7fd63a3cb6ad0cdfa627cec46a8
Thanks for your help!